### PR TITLE
CVE-2017-5052

### PR DESCRIPTION
**Pull request by wrs1492**
Vulnerability in the rendering engine that caused memory allocation issues when blocks were nested in each other.

### CODE REVIEW

File: cves/CVE-2017-5052.yml [hunk 1]
```diff
@@ -80,9 +81,11 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: There were automated unit tests in this vulnerability and the original
```
Comment (mesternefeld) [inline cves/CVE-2017-5052.yml]: This should help fix the CI
```suggestion
  answer: |
    There were automated unit tests in this vulnerability and the original
    code was unit tested. The fix did not involve improving the automated tests. 
  code: |
    The bug was an extremely niche situation, so while the code was unit tested,
    it did not get the use case that caused the vulnerability.
    fix: A unit test was added for this issue. 
```

File: cves/CVE-2017-5052.yml [hunk 2]
```diff
@@ -201,4 +206,9 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: It is hard to say what type of mistake this was as it was such a niche
+  vulnerability on a wierd edge case that ended up having very serious security
+  implications. I would have to say this falls under a coding mistake as the
```
Comment (bjt6087) [inline cves/CVE-2017-5052.yml]: "weird edge case"

**Comment by bjt6087** [comment]
Would CWE-119 fit this vulnerability?

**Comment by bjt6087** [comment]
Interesting vulnerability. 3 upvotes

File: cves/CVE-2017-5052.yml [hunk 3]
```diff
@@ -98,20 +101,22 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: The vulnerability was discoverd by a misuse case, which caused chrome to crash
```
Comment (Ramenous) [inline cves/CVE-2017-5052.yml]: If possible, can you describe and elaborate on what was the misuse case that caused chrome to crash? Providing the proper use case along with the misuse case can also help differentiate what was the difference in flow that caused this vulnerability

File: cves/CVE-2017-5052.yml (hunk 2 above)
Comment (TLezynski) [reply cves/CVE-2017-5052.yml]: Are you certain it was a coding mistake? If this is an edge case, perhaps the edge case wasn't considered in the design and it's a design flaw.

File: cves/CVE-2017-5052.yml (hunk 3 above)
Comment (Ramenous) [inline cves/CVE-2017-5052.yml]: If possible, can you describe and elaborate on what was the misuse case that caused chrome to crash? Providing the proper use case along with the misuse case can also help differentiate what was the difference in flow that caused this vulnerability

File: cves/CVE-2017-5052.yml (hunk 2 above)
Comment (wrs1492) [reply cves/CVE-2017-5052.yml]: I would not say it was a design flaw as the fix for the bug was a relatively simple change, and took care of the problem thoroughly. Going through the history of the bug, it was really hard to produce a test case that would reliably cause the bug to occur, so I would say it is more of a testing mistake than a coding one, so I will update the file accordingly

File: cves/CVE-2017-5052.yml (hunk 3 above)
Comment (wrs1492) [reply cves/CVE-2017-5052.yml]: Yeah sorry I ran out of time when submitting this vulnerability, the crash was caused by nested "blocks", and I will flesh out this answer.

**Comment by wrs1492** [comment]
> line 7: I see that you didn't add a CWE for this vulnerability. Is that intentional in that you couldn't find an appropriate one? or did you miss it?

I ran out of time when submitting, I will find an appropriate one

**Comment by wrs1492** [comment]
> CWE-119

I think this might be a good fit, I will see if there are any others that fit better. Thanks for the help.

**Review by rlb8800 — APPROVED** [review]

**Comment by rlb8800** [comment]
Fixed this on dev

**Pull request closed without merging**